helpers.go

```diff
@@ -2,16 +2,34 @@ package psrpc
 
 import (
 	"math/rand"
+	"sync"
 	"time"
 	"unicode"
 
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
 )
 
-const alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+type lockedRandSource struct {
+	mu  sync.Mutex
+	src rand.Source
+}
+
+func (s *lockedRandSource) Int63() int64 {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.src.Int63()
```

Comment:
Was this racing?

Reply:
yeah, i assumed this was thread safe but it panicked while running the cloud load tester locally. this is how the global rng is protected in math/rand more or less https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/math/rand/rand.go;l=304